src/Microsoft.Build.Sql.Templates/sqlproject/SqlProject1.sqlproj

```diff
@@ -1,9 +1,16 @@
-<?xml version="1.0" encoding="utf-8"?>
-<Project DefaultTargets="Build">
-  <Sdk Name="Microsoft.Build.Sql" Version="###ASSEMBLY_VERSION###" />
-  <PropertyGroup>
-    <Name>SqlProject1</Name>
-    <DSP>Microsoft.Data.Tools.Schema.Sql.{TargetPlatform}DatabaseSchemaProvider</DSP>
-    <ModelCollation>1033, CI</ModelCollation>
-  </PropertyGroup>
-</Project>
\ No newline at end of file
+<Project Sdk="Microsoft.Build.Sql/###ASSEMBLY_VERSION###" DefaultTargets="Build">
+
+    <PropertyGroup>
+        <SqlTargetName>SqlProject1</SqlTargetName>
+        <DacVersion>1.0.0.0<DacVersion>
+        <DSP>Microsoft.Data.Tools.Schema.Sql.{TargetPlatform}DatabaseSchemaProvider</DSP>
+        <ModelCollation>1033, CI</ModelCollation>
+    </PropertyGroup>
+
+    <!-- Correct the build in Visual Studio so it doesn't trigger a NuGet Framework Reference error.
+    <Target Name="PreBuild" BeforeTargets="PreBuildEvent" Condition="'$(MSBuildRuntimeType)' != 'Core'">
+        <Message Importance="high" Text="Ensuring the $(MSBuildThisFileDirectory)\obj\project.assets.json file is removed, if necessary, so that the database project can be built through VisualStudio SSDT without errors" />
+        <Delete Files="$(MSBuildThisFileDirectory)\obj\project.assets.json" />
```

Comment:
If I understand correctly, this target would be triggered for every build. @Ri7Sh Do you see any issues with that?

Reply:
@robertmclaws Can you please raise an issue describing the bug as well?

Reply:
The code originally came from this post: https://github.com/microsoft/DacFx/issues/180#issuecomment-2171445288 and fixes issue #180.

If you think it would be better in the SDK.targets file, I'm happy to do another check-in that moves it there.

Reply:
ah - clarifying in this thread - this doesn't fix support in VS, it fixes the specific case in VS when you try to build the project after having built it on the same machine with .NET 8 (cli or VS Code). This would be good to resolve @Ri7Sh but let's make sure we converge on an approach in the .NET template as well as in the VS template.

Reply:
@dzsquared I believe that is correct. For more context, if you build in VS the first time, it will build correctly. 

But let's say you're trying to verify the build on the CLI so you're sure it will run in your CI/CD pipeline.

The next time you will build in Visual Studio, you will get the following error:

`Your project does not reference ".NETFramework,Version=v4.7.2" framework. Add a reference to ".NETFramework,Version=v4.7.2" in the "TargetFrameworks" property of your project file and then re-run NuGet restore.`

But that is not explicitly a .NET 8 problem, as this happened while I was trying to move to an SDK-style project in a .NET 9 RC2 app. I personally believe it is an issue in the downstream targets file that sneaks in a .NET Framework reference to get the compiler to work, which I believe is a mistake. But I can't find where to contribute a fix to that targets file.

Including this in the project template makes sure that error doesn't happen.

if you would like I can take this out of this PR and put it in a separate one... however this PR contained everything I needed to get a build working consistently both locally and in Azure DevOps, which is why this fix was also included here so it could get into customer's hands ASAP during the preview.

HTH!